_source/_posts/2021-10-08-private-key-jwt-python.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: blog_post
+title: "Set Up the Private Key JWT Flow in Three Python Commands"
+author: cale-switzer
+by: advocate
+communities: [python]
+description: "The Private Key JWT flow is one of the more complicated OAuth flows to set up. Learn how the flow works at a high level and how you can implement it in three commands with Python scripting."
+tags: [python, oauth, private-key-jwt]
+image:
+type: awareness
+---
+
+The [Private Key JWT flow](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication) is one of the more complicated OIDC flows to set up. If you already know HOW it works and just want to get to the code, [click here](#python-private-key-jwt-setup) to skip ahead. If you're looking to gain a better understanding of how it works, here's a high level explanation:
+
+{% img blog/private-key-jwt-python/pkj_1.png alt:"Private Key JWT FLowchart" width:"1000" %}{: .center-image }
+
+Once you've obtained the access token from this flow, you can throw it into the Authorization header of requests to your Okta APIs:
+
+{% img blog/private-key-jwt-python/pkj_2.png alt:"Use Access Token" width:"400" %}{: .center-image }
+
+But let's break it down a bit more:
+
+### Step 1
+
+Typically, one may think of JWTs as strictly being in the realm of the Identity Provider (IdP) - and in most cases that's correct. When an IdP grants tokens to a service provider (SP), it is signing those tokens with a private/secret key. Part of the responsibility of the SP when validating tokens is to use a public key provided by the IdP to verify the signature.
```

Comment:
```suggestion
Typically, one may think of JWTs as strictly being in the realm of the *identity provider* (IdP) - and in most cases that's correct. When an IdP grants tokens to a service provider (SP), it is signing those tokens with a private/secret key. Part of the responsibility of the SP when validating tokens is to use a public key provided by the IdP to verify the signature.
```
using italics for first mention of "identity provider" instead of uppercase